cmd/swapd/main.go

```diff
@@ -643,17 +648,26 @@ func newBackend(
 		return nil, err
 	}
 
-	// For the monero wallet related values, keep the default config values unless the end
-	// use explicitly set the flag.
-	if c.IsSet(flagMoneroDaemonHost) {
-		cfg.MoneroDaemonHost = c.String(flagMoneroDaemonHost)
-		if cfg.MoneroDaemonHost == "" {
-			return nil, errFlagValueEmpty(flagMoneroDaemonHost)
+	if c.IsSet(flagMoneroDaemonHost) || c.IsSet(flagMoneroDaemonPort) {
+		node := &common.MoneroNode{
+			Host: "127.0.0.1",
+			Port: common.DefaultMoneroPortFromEnv(env),
 		}
+		if c.IsSet(flagMoneroDaemonHost) {
+			node.Host = c.String(flagMoneroDaemonHost)
+			if node.Host == "" {
+				return nil, errFlagValueEmpty(flagMoneroDaemonHost)
+			}
+		}
+		if c.IsSet(flagMoneroDaemonPort) {
+			node.Port = c.Uint(flagMoneroDaemonPort)
+			if node.Port == 0 {
+				return nil, errFlagValueZero(flagMoneroDaemonPort)
+			}
+		}
+		cfg.MoneroNodes = []*common.MoneroNode{node}
```

Comment:
do you think it would be useful to allow users to specify multiple monero daemons via cli? I guess since we only end up using one anyways, it probably isn't super useful.

Reply:
Hopefully, people are using the flag to specify their own local server, in which case it wouldn't be a useful feature.  If we figure out that it is a valid use case in the future, I think we'd need to switch to a host:port syntax that possibly defaults the port if no colon is given.